frontend/src/metabase/visualizations/components/legend/LegendCaption.jsx

```diff
@@ -33,16 +35,48 @@ function shouldHideDescription(width) {
   return width != null && width < HIDE_DESCRIPTION_THRESHOLD;
 }
 
+/**
+ * Using non-empty href will ensure that a focusable link is rendered.
+ * We need a focusable element to handle onFocus.
+ * (Using a div with tabIndex={0} breaks the sequence of focusable elements)
+ */
+const HREF_PLACEHOLDER = "#";
+
 const LegendCaption = ({
   className,
   title,
   description,
-  href,
+  getHref,
   icon,
   actionButtons,
   onSelectTitle,
   width,
 }) => {
+  /*
+   * Optimization: lazy computing the href on title focus & mouseenter only.
+   * Href computation uses getNewCardUrl, which makes a few MLv2 calls,
+   * which are expensive.
+   * It's a performance issue on dashboards that have hundreds of dashcards
+   * (during initial render and after changing dashboard parameters which can
+   * potentially affect the href).
+   */
+  const [href, setHref] = useState(getHref ? HREF_PLACEHOLDER : undefined);
+  const store = useStore();
+
+  const handleFocus = useCallback(() => {
+    if (getHref) {
+      const state = store.getState();
```

Comment:
Can we move the store part to `DashCard`? And pass a callback that doesn't need any arguments